enhancements/sig-architecture/70-spread-policy/README.md

```diff
@@ -0,0 +1,226 @@
+  - topologyKey: region
+    topologyKeyType: Label
+    weight: 10
+  # I plan to exclude the spread policy in Additive mode.
+  # In Exact mode, users need to specify the weight of them manually.
+  # The spread policy only work when there is a SpreadPolicy coordinate
+  # in prioritizerPolicy.
+  prioritizerPolicy:
+    mode: Exact
+    configurations:
+      - scoreCoordinate:
+          builtIn: ResourceAllocatableMemory
+      - scoreCoordinate:
+          builtIn: SpreadPolicy
+        weight: 2
+```
+
+### Implementation
+The current scheduling framework contains two types of plugins, i.e., `Filter`s and `Prioritizer`s. 
+`Filter`s are invoked *before* `Prioritizer`s to express hard constrains, filtering the clusters. 
+`Prioritizer`s express soft constraints, scoring the clusters. The spread policy can be a hard constraint (when `maxSkew` is not omitted for some terms) or a soft constraint (when `maxSkew` is omitted for all terms).
+For a hard spread constraint, the scheduler need to perform the hard decision logic (i.e., decide to schedule or not according to `maxSkew`) *after* scoring the clusters, which is unable to implement in a `Filter`.
+Besides, we cannot just implement the hard decision logic in a `Prioritizer` in some way.
+The reason is that each `Prioritizer` score the clusters individually, while the `maxSkew` constraint should apply to the final scheduling result, which can be affected by all the `Prioritizer`s.
+
+Hence, a new plugin type, `Selector`, is proposed to express hard constrains *after* `Prioritizer`s.
+The `Selector` interface is showed below.
+
+```go
+// Selector defines a selector plugin that selects clusters after scoring them.
+type Selector interface {
+	Plugin
+
+	// Select returns the selected clusters.
+	Select(ctx context.Context, placement *clusterapiv1beta1.Placement, clusters []*clusterapiv1.ManagedCluster, scores map[string]int64) PluginSelectResult
+}
+
+// PluginSelectResult contains the details of a select plugin result.
+type PluginSelectResult struct {
+	// Selected contains the selected ManagedCluster.
+	Selected []*clusterapiv1.ManagedCluster
+	// Err contains the select plugin error message.
+	Err error
+}
+```
+
+Then, a `EvenSpreadSelector` will be implemented as follows to select n clusters.
+
+1. The result cluster set = {}
+2. Call `selectOne` n times, in each time insert the returned cluster into the result cluster set
+3. Return the result cluster set
+
+`selectOne`:
+
+1. For each cluster, calculate a score (even_score) according the benefit of selection of it to balancing the skewness. If the selection of a cluster would cause the violation to `maxSkew`, exclude it from the later selecting phases. If all clusters are excluded, report a error.
+2. For each cluster, calculate its final score = u*even_score + its_score_from_prioritizers, where u is a parameter.
+3. Sort the clusters by final scores and return the one with the highest final score.
```

Comment:
let's define the graduation criteria here.

Reply:
Done.